namespace.go

```diff
@@ -94,6 +94,10 @@ func (n *SimpleNamespace) RemoveExperiment(name string) error {
 		}
 	}
 
+	for _, val := range segmentsToFree {
+		delete(n.segmentAllocations, uint64(val))
```

Comment:
Could `delete` be done inside the loop where values are added to`segmentsToFree`? (instead of iterating `segmentsToFree` outside separately)
https://github.com/biased-unit/planout-golang/blob/master/namespace.go#L98